sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2476,7 +2500,7 @@ object GpuOverrides {
       "The backend for hash based aggregations",
       ExecChecks(
         (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.MAP)
-            .nested(TypeSig.STRING),
+            .nested(TypeSig.STRING) + TypeSig.ARRAY,
         TypeSig.all),
```

Comment:
@revans2 this is an additional change per observations from a sample _user_ query I ran for array_contains. I want an array to be a valid input but not be nested in any way and that is what I am going for here. Let me know if you think I should do it separately or differently.

Reply:
The way this is set up it will only work for an array of Strings. Is that what you wanted?

Reply:
Oh! Is that so? The 'ARRAY` is outside the nested call where we say we only support strings so I thought that would not be the case. 
For arrays of basic types but not decimals and other nested types , would this
```
      ExecChecks(
        (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.MAP)
            .nested(TypeSig.STRING) + TypeSig.ARRAY.nested(TypeSig.commonCudfTypes),
        TypeSig.all)
```
make more sense? I want to support arrays of basic types + arrays of strings aka `commonCudfTypes`

Reply:
I have made some changes here that might be more relevant based on the docs they seem to generate. Also, tested on the sample query to make sure that works. Appreciate any thoughts on this @revans2

Reply:
My only concern now is that it looks like we are restricting the checks in a way that they do not support.

You only get top level types and nested types.  You cannot say that a Map of Strings or an Array of Doubles is OK, but nothing else.  You would end up with a Map of Strings, or Doubles OR and Array of Strings or Doubles.  If we run into a situation where we have to have something like that we can look at adding it in.  But until then I think it  would be best to just have it look more like.
```
(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.MAP + TypeSig.ARRAY)
  .nested(TypeSig.commonCudfTypes + TypeSig.NULL)
```

Reply:
Thanks Bobby. Makes sense, I was hoping to keep existing one as is and tack this on. Using the suggested approach.